GithubApp/Controllers/FilterViewController.swift

```diff
@@ -9,27 +9,90 @@ import UIKit
 
 class FilterViewController: UIViewController {
 
+    //MARK: Instance Properties
+    var delegate: FilterDelegate?
+    var filterData :[Filter] = []
+    let checkedImage = UIImage(named: "check")! as UIImage
+    let uncheckedImage = UIImage(named: "uncheck")! as UIImage
+
+    // MARK: IB outlet
     @IBOutlet var filterView: FilterView!
+    
     override func viewDidLoad() {
-        super.viewDidLoad()
+        //super.viewDidLoad()
+        print("View Did load 2 called")
         filterView.filterTableView.delegate = self
         filterView.filterTableView.dataSource = self
     }
-    let languagesData = ["Python","Interior Painting","Electrician","Inappropriate client behavior","Gardener","Carpenter","House Cleaning"]
+    
+    // MARK: Functions
+    func configure(_ filterData: [Filter]){
+        self.filterData.removeAll()
+        self.filterData = filterData
+    }
+
+    // MARK: IB Actions
+    @IBAction func clearFilter(_ sender: Any) {
+        for index in 0..<filterData.count{
+            filterData[index].isSelected = false
+        }
```

Comment:
you can avoid for loop by doing this
```
filterData = Array(repeating: false, count: filterData.count)
```

Reply:
this cannot be done using arrays but we can do
filterData.indices.foreach{
filterData[indices].isSelected = false
}

Reply:
have you tried it.